### DISCUSSION TITLE
Magic Discussions

### DISCUSSION THREAD

**Discussion by RTimothyEdwards** [RTimothyEdwards/magic#124] [Announcements]
## Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DalinEM-lab** [comment]
Hi 
I have installed IHP-Open-PDK (sg13g2) and integrated with Magic. However, importing spice file option is not there. any idea how to solve this issue.
![magic](https://github.com/user-attachments/assets/6f7122a8-bedc-4f56-a2c6-86f6f68f8328)

**Comment by RTimothyEdwards** [comment]
@DalinEM-lab:  the "Import SPICE" function requires that there are device generators (a.k.a. pcells).  The support for magic in IHP SG13G2 is an ongoing project.  The 2nd part of that project is to create all of the generator cells.  That work was supposed to have been finished in mid-April, but with the shutdown of Efabless, it will likely be delayed a bit, maybe another month, but the work will definitely continue.

**Reply by DalinEM-lab** [reply]
@RTimothyEdwards  I’ve used the available TCL script, but it may not be usable at the moment. I’m looking forward to working with the IHP PDK and Magic VLSI in the future. Thank you for your efforts.
![magic_latest](https://github.com/user-attachments/assets/f24206ef-3d53-4775-bf35-ffc0dcb45440)

**Comment by bhagwat-rahul** [comment]
Wondering if there have been any updates with this and what the community can do to help

**Comment by RTimothyEdwards** [comment]
@bhagwat-rahul :  Well, sure.  IHP SG13G2 and SG13CMOS5L support for magic was merged into the IHP upstream repository some time ago.  I used it to tape out a test chip on the 1st SG13CMOS5L shuttle run (https://github.com/RTimothyEdwards/sg13cmos5l_ocd_openframe) with a handful of analog components.  This should be included in the (default) IHP support for IIC-OSIC-TOOLS as well.  Is there anything in particular that I can point you to?

**Reply by bhagwat-rahul** [reply]
I was curious since i didn't see the Import SPICE option for ihp-sg13g2 (running magic from within IIC OSIC tools). Was on a mid 2025 version of iic tools, recreating latest image to see if it works.

**Reply by bhagwat-rahul** [reply]
Update: This works! Just needed to update iic-tools (was on 2025.07 upgraded to latest)

**Reply by RTimothyEdwards** [reply]
Good to know, thanks!

**Comment by RTimothyEdwards** [comment]
I should mention that while the contract with Efabless ended abruptly along with Efabless, IHP was very generous in creating a new contract which allowed me and Leo Moser to finish the work.  It was about half a year late from the original plan, but it got done.
